pkg/init/backend/interface.go

```diff
@@ -21,8 +21,9 @@ type InitBackend interface {
 	// depending on the flags. If not starter project is selected, a nil starter is returned
 	SelectStarterProject(devfile parser.DevfileObj, flags map[string]string) (starter *v1alpha2.StarterProject, err error)
 
-	// PersonalizeName updates a devfile name, depending on the flags
-	PersonalizeName(devfile parser.DevfileObj, flags map[string]string) error
+	// PersonalizeName updates a devfile name, depending on the flags.
+	// It optionally writes the resulting Devfile to disk depending on `writeToDisk`.
+	PersonalizeName(devfile *parser.DevfileObj, flags map[string]string, writeToDisk bool) error
```

Comment:
I would prefer to not have the `writeToDisk` option, and make the method never write to disk. The caller would be responsible for writing to disk in every case (that would remove the filesystem dependency on the method). Also, as we are modifying only the `Data` field of the devfileObj, we could pass it only.

Reply:
Done. See https://github.com/redhat-developer/odo/pull/5553#issuecomment-1068159050